app/logic/getUpcomingEvents.py

```diff
@@ -0,0 +1,29 @@
+from app.models.program import Program
+from app.models.interest import Interest
+from app.models.event import Event
+from app.models.user import User
+from datetime import date, datetime, time
+
+def getUpcomingEventsForUser(username):
+    user = User.get(User.username == username)
+    interestedEvent = (Event.select(Event, Program, Interest)
+                            .join(Program)
+                            .switch(Event)
+                            .join(Interest, on=(Event.program == Interest.program))
+                            .where(Event.program_id == Program.id, Interest.user == user)

```

Comment:
The first part of this WHERE clause is unecessary, since you already implicitly have a join condition between those two tables.

Also, I think .switch is also unnecessary, as you are explicitly providing the join condition when you join Interest

Reply:
We removed the first part of this WHERE clause and .switch and the code still works.

Reply:
We also removed the showUpcomingEvent.py file and call getUpcomingEventsForUser directly from routes.py